### PR TITLE
Added Frontend numpy random triangular function

### LINKED ISSUE #18561: triangular

### PR DESCRIPTION
**Pull request by VedantPol**
Close #18561

### CODE REVIEW

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Comment by VedantPol** [comment]
@bipinKrishnan would love to get this PR reviewed :)

File: ivy/functional/frontends/numpy/random/functions.py [hunk 1]
```diff
@@ -183,3 +183,17 @@ def gumbel(loc=0.0, scale=1.0, size=None):
     u = ivy.random_uniform(low=0.0, high=1.0, shape=size, dtype="float64")
     x = loc - scale * ivy.log(-ivy.log(u))
     return x
+
+
+@to_ivy_arrays_and_back
+@from_zero_dim_arrays_to_scalar
+def triangular(left, mode, right, size=None):
+    if left > mode or mode > right or left == right:
+        raise ValueError("left < mode < right is not being followed")
```
Comment (bipinKrishnan) [inline ivy/functional/frontends/numpy/random/functions.py:192]: May be replace this with `IvyValueError`

**Comment by bipinKrishnan** [comment]
Feels like the CI did not run the new tests you added, I'll check this with the team and get back to you :slightly_smiling_face:

File: ivy/functional/frontends/numpy/random/functions.py (hunk 1 above)
Comment (VedantPol) [reply ivy/functional/frontends/numpy/random/functions.py]: replaced ValueError with IvyValueError

**Comment by VedantPol** [comment]
hey @bipinKrishnan ,
I am getting the following error while running the test locally. If I import the file and use the following test case inputs it works fine. Not sure why I am getting this error. Hope you could help me out :)

ivy_tests/test_ivy/test_frontends/test_numpy/test_random/test_functions.py:711 (test_numpy_triangular[cpu-ivy.functional.backends.numpy-False-False])
frontend = 'numpy', on_device = 'cpu'

    @handle_frontend_test(
>       fn_tree="numpy.random.triangular",
        input_dtypes=helpers.get_dtypes("float"),
        left=st.floats(
            allow_nan=False,
            allow_infinity=False,
            width=32,
            min_value=0,
            max_value=10,
        ),
        mode=st.floats(
            allow_nan=False,
            allow_infinity=False,
            width=32,
            min_value=10,
            max_value=100,
            exclude_min=True,
        ),
        right=st.floats(
            allow_nan=False,
            allow_infinity=False,
            width=32,
            min_value=100,
            max_value=1000,
            exclude_min=True,
        ),
        size=helpers.get_shape(allow_none=False),
    )

test_functions.py:713: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
test_functions.py:751: in test_numpy_triangular
    helpers.test_frontend_function(
../../../helpers/function_testing.py:604: in test_frontend_function
    ret = get_frontend_ret(
../../../helpers/function_testing.py:1776: in get_frontend_ret
    ret = frontend_fn(*args, **kwargs)
/opt/project/ivy/functional/frontends/numpy/func_wrapper.py:404: in _outputs_to_frontend_arrays
    ret = fn(*args, **kwargs)
/opt/project/ivy/functional/frontends/numpy/func_wrapper.py:356: in _inputs_to_ivy_arrays_np
    return fn(*ivy_args, **ivy_kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

args = ((),), kwargs = {'left': 0.0, 'mode': 11.0, 'right': 101.0}

    @functools.wraps(fn)
    def _from_zero_dim_arrays_to_scalar(*args, **kwargs):
        """
        Convert 0 dimensional arrays to float numbers.
    
        Call the function, and then converts all 0 dimensional array instances in the
        function to float numbers if out argument is not provided.
    
        Parameters
        ----------
        args
            The arguments to be passed to the function.
    
        kwargs
            The keyword arguments to be passed to the function.
    
        Returns
        -------
            The return of the function, with 0 dimensional arrays as float numbers.
        """
        # call unmodified function
>       ret = fn(*args, **kwargs)
E       TypeError: triangular() got multiple values for argument 'left'
E       Falsifying example: test_numpy_triangular(
E           frontend='numpy',
E           on_device='cpu',
E           input_dtypes=['float16', 'float64', 'float32'],
E           left=0.0,
E           mode=11.0,
E           right=101.0,
E           size=(),
E           test_flags=FrontendFunctionTestFlags(
E               num_positional_args=1,
E               with_out=False,
E               inplace=False,
E               as_variable=[False],
E               native_arrays=[False],
E               generate_frontend_arrays=False,
E           ),
E           fn_tree='ivy.functional.frontends.numpy.random.triangular',
E       )
E       
E       You can reproduce this example by temporarily adding @reproduce_failure('6.81.1', b'AXicY2DAArgRzFQGBkYgBQADtABy') as a decorator on your test case

**Comment by bipinKrishnan** [comment]
> hey @bipinKrishnan , I am getting the following error while running the test locally. If I import the file and use the following test case inputs it works fine. Not sure why I am getting this error. Hope you could help me out :)
> 
> ivy_tests/test_ivy/test_frontends/test_numpy/test_random/test_functions.py:711 (test_numpy_triangular[cpu-ivy.functional.backends.numpy-False-False]) frontend = 'numpy', on_device = 'cpu'
> 
> ```
> @handle_frontend_test(
> ```
> 
> > ```
> >   fn_tree="numpy.random.triangular",
> > ```
> 
> ```
>     input_dtypes=helpers.get_dtypes("float"),
>     left=st.floats(
>         allow_nan=False,
>         allow_infinity=False,
>         width=32,
>         min_value=0,
>         max_value=10,
>     ),
>     mode=st.floats(
>         allow_nan=False,
>         allow_infinity=False,
>         width=32,
>         min_value=10,
>         max_value=100,
>         exclude_min=True,
>     ),
>     right=st.floats(
>         allow_nan=False,
>         allow_infinity=False,
>         width=32,
>         min_value=100,
>         max_value=1000,
>         exclude_min=True,
>     ),
>     size=helpers.get_shape(allow_none=False),
> )
> ```
> 
> test_functions.py:713:
> 
> test_functions.py:751: in test_numpy_triangular helpers.test_frontend_function( ../../../helpers/function_testing.py:604: in test_frontend_function ret = get_frontend_ret( ../../../helpers/function_testing.py:1776: in get_frontend_ret ret = frontend_fn(*args, **kwargs) /opt/project/ivy/functional/frontends/numpy/func_wrapper.py:404: in _outputs_to_frontend_arrays ret = fn(*args, **kwargs) /opt/project/ivy/functional/frontends/numpy/func_wrapper.py:356: in _inputs_to_ivy_arrays_np return fn(*ivy_args, **ivy_kwargs)
> 
> args = ((),), kwargs = {'left': 0.0, 'mode': 11.0, 'right': 101.0}
> 
> ```
> @functools.wraps(fn)
> def _from_zero_dim_arrays_to_scalar(*args, **kwargs):
>     """
>     Convert 0 dimensional arrays to float numbers.
> 
>     Call the function, and then converts all 0 dimensional array instances in the
>     function to float numbers if out argument is not provided.
> 
>     Parameters
>     ----------
>     args
>         The arguments to be passed to the function.
> 
>     kwargs
>         The keyword arguments to be passed to the function.
> 
>     Returns
>     -------
>         The return of the function, with 0 dimensional arrays as float numbers.
>     """
>     # call unmodified function
> ```
> 
> > ```
> >   ret = fn(*args, **kwargs)
> > ```
> 
> E TypeError: triangular() got multiple values for argument 'left' E Falsifying example: test_numpy_triangular( E frontend='numpy', E on_device='cpu', E input_dtypes=['float16', 'float64', 'float32'], E left=0.0, E mode=11.0, E right=101.0, E size=(), E test_flags=FrontendFunctionTestFlags( E num_positional_args=1, E with_out=False, E inplace=False, E as_variable=[False], E native_arrays=[False], E generate_frontend_arrays=False, E ), E fn_tree='ivy.functional.frontends.numpy.random.triangular', E ) E E You can reproduce this example by temporarily adding @reproduce_failure('6.81.1', b'AXicY2DAArgRzFQGBkYgBQADtABy') as a decorator on your test case

The issue is that in the `test_flags` arguments mentioned here, the number of positional argument expected is 1, but all of the arguments are provided as kwargs, making `num_positional_args=0` should solve the issue probably.

**Comment by VedantPol** [comment]
hey @bipinKrishnan ,
Thanks for the hint. All of my local tests are passing. But for some reason, a few of the GitHub action tests failed. Would like to hear your take on the following :) 
Screenshot from 2023-07-24 22-05-59

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_random/test_functions.py [hunk 2]
```diff
@@ -784,3 +784,58 @@ def test_numpy_gamma(
...
+    ),
+    right=st.floats(
+        allow_nan=False,
+        allow_infinity=False,
+        width=32,
+        min_value=100,
+        max_value=1000,
+        exclude_min=True,
+    ),
+    size=helpers.get_shape(allow_none=False),
```
Comment (bipinKrishnan) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_random/test_functions.py:815]: May be `allow_none` should be set to True given that the default values of size is None

Comment (VedantPol) [reply ivy_tests/test_ivy/test_frontends/test_numpy/test_random/test_functions.py]: updated it , thanks

**Comment by bipinKrishnan** [comment]
Lgtm! Thanks for the contribution @VedantPol :rocket:

**Pull request merged**